src/pages/Create.vue

```diff
@@ -0,0 +1,200 @@
+        <ion-col>Quamtity</ion-col>
+        <ion-col>Unit Price</ion-col>
+        <ion-col>Sub-Total</ion-col>
+      </ion-row>
+      <ion-row v-for="item in itemDetails" :key="item.name">
+        <ion-col>{{ item.name }}</ion-col>
+        <ion-col>{{ item.quantity }}</ion-col>
+        <ion-col>{{ item.unitPrice }}</ion-col>
+        <ion-col>{{ item.subTotal }}</ion-col>
+      </ion-row>
+      <form @submit.prevent="addItemDetail">
+        <ion-row>
+          <ion-input
+            type="text"
+            autofocus
+            v-model="item.name"
+            name="name"
+            placeholder="item"
+            required
+          ></ion-input>
+          <ion-input
+            type="number"
+            autofocus
+            v-model="item.quantity"
+            name="quantity"
+            placeholder="Quantity"
+            required
+          ></ion-input>
+          <ion-input
+            type="number"
+            autofocus
+            v-model="item.unitPrice"
+            name="unitPrice"
+            placeholder="Price"
+            required
+          ></ion-input>
+          <ion-input
+            type="number"
+            autofocus
+            v-model="item.subTotal"
+            name="subTotal"
+            placeholder="Total Price"
+            disabled
+          ></ion-input>
+        </ion-row>
+        <ion-button button type="submit">ADD</ion-button>
+      </form>
+      <ion-text color="light">
+        <h3>Grand Total {{orderDetail.grandTotal}}</h3>
+      </ion-text>
+      <ion-button class="submitButton" button type="submit">Create</ion-button>
+      <ion-button class="cancelButton" button href="/home">Cancel</ion-button>
+    </form>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import { IonContent, IonTitle, IonItem, IonInput } from "@ionic/vue";
+import { defineComponent, ref } from "vue";
+import Toolbar from "@/shared/components/Toolbar.vue";
+import IItem from "@/core/interfaces/item";
+import IOrderDetail from "@/core/interfaces/order-detail"
+
+export default defineComponent({
+  name: "Create",
+  components: {
+    IonContent,
+    Toolbar,
+    IonTitle,
+    IonItem,
+    IonInput,
+  },
+  setup() {
+    const item = ref<IItem>({
+      name: "",
+      quantity: 0,
+      unitPrice: 0,
+      subTotal: 0,
+    });
+
+    const itemDetails = ref<IItem[]>([]);
+
+    const orderDetail = ref<IOrderDetail>({
+      item: [{
+        name: "",
+        quantity: 0,
+        unitPrice: 0,
+        subTotal: 0,
+      }],
+      supplier: "",
+      purchaseDate: new Date,
+      grandTotal: 0,
+      description: ""
+    });
+
+    return { itemDetails, item, orderDetail };
+
+  },
+  methods: {
+    addItemDetail() {
+      this.item.subTotal = this.item.unitPrice * this.item.quantity;
+      this.itemDetails.push(this.item);
+      
+      this.item = {
+        name: "",
+        quantity: 0,
+        unitPrice: 0,
+        subTotal: 0,
+      };
```

Comment:
Initialize to null if want to have a "nothing"

Reply:
done